platform/app-manager/app-manager.js

```diff
@@ -2,87 +2,132 @@ const codeContainer = require(`${__dirname}/code-container/container`);
 const appsDb = 'apps';
 const appsInfoCollection = 'info';
 
 // Initialize database connection once
-var db;
+let db;
 mongoClient.connect(mongoUrl, { useNewUrlParser: true }, function(err, client) {
-    if(err) throw err;
+    if(err) {
+        console.error("[ERROR] Failed to connect to mongodb.");
+        throw err;
+    }
     db = client.db(appsDb);
 });
 
-// Create logs directory if not present
+// TODO: When initializing app-manager, app-manager checks database to see 
+// if any apps already existed (zombie process). This happens when app-manager crashed abnormally
+
+// Create logs directory for apps if not present
 fs.ensureDirSync(`${__dirname}/logs`);
 /**
  * This function saves the app info to the database
- * @param {string} appPath application path
  * @param {string} topic mqtt topic
- * @param {string} pid application pid
- * @returns {string}
+ * @param {string} appPath application executable path
+ * @param {string} metadataPath metadata path
+ * @param {string} pid application's pid
  */
-function saveAppInfoToDB(appPath, topic, pid) {
-    let appId;
-    try {
-        let result = db.collection(appsInfoCollection).insertOne( { "appName": appPath, "topic": topic, "pid": pid });
-        appId = result["insertedId"];
-    } catch (err) {
-        console.error(err);
+function saveAppInfoToDB(topic, appPath, metadataPath, pid) {
+    let result = db.collection(appsInfoCollection)
+                    .insertOne({
+                        "name": path.basename(appPath),
+                        "topic": topic,
+                        "pid": pid,
+                        "pathPath": appPath,
+                        "metadataPath": metadataPath
+                    });
+    if("writeConcernError" in result) {
+        throw result["writeConcernError"];
     }
-    return appId;
 }
 
-// apps stores process, topic, pid, and path
+// Stores the process, topic, pid, appPath, and metadataPath in apps
+// apps = {
+//     "app-name": {
+//         "app": instance-of-process,
+//         "pid": application-process-pid,
+//         "topic": topic, // application's id
+//         "appPath": application-executable-path,
+//         "metadataPath": application-metadata-path
+//     }
+// }
 const apps = {};
 
 /**
  * This function generates the topic for new coming app.
- * TODO: use app name or other?
- * @param {string} appName
+ * the topic is the id generated by mongodb
+ * @param {string} appPath
  * @returns {string}
  */
-function getTopic(appName) {
-    let topic = appName;
-    return topic;
+function generateTopic(appPath) {
+    // The fastest algorithm is sha1-base64
+    // Reference: https://medium.com/@chris_72272/what-is-the-fastest-node-js-hashing-algorithm-c15c1a0e164e
+    const hash = crypto.createHash('sha1');
+    // use the timestamp to create topic
+    let dirName = path.dirname(appPath).split(path.sep).pop();
+    hash.update(dirName);
+    return hash.digest('base64');
```

Comment:
Let's just avoid all dependencies on the codeContainer directory name.

Instead, to be sure, just hash on the current_TS **generated here** and the appName. 

do a hash(appName + Date.now()) to ensure we don't have an issue when appName of two apps are the same.